content/200-concepts/100-components/01-prisma-schema/06-relations/420-relation-mode.mdx

```diff
@@ -0,0 +1,214 @@
+
+### Which referential integrity checks are emulated?
+
+When you _update_ a record, Prisma will emulate referential integrity checks. This means that when you update a record with a relation to another record, the related record needs to exist. If the record does not exist, Prisma Client will return an error message.
+
+However, when you _create_ a record, Prisma does not emulate any referential integrity checks. You will be able to create invalid data.
+
+### Which referential actions are emulated?
+
+When you _update_ or _delete_ a record with related records, Prisma will emulate referential actions.
+
+<!-- TODO: list actions with brief description? -->
+
+The following table shows which emulated referential actions are available for each database connector:
+
+| Database    | Cascade | Restrict | NoAction | SetNull | SetDefault |
+| :---------- | :------ | :------- | :------- | :------ | :--------- |
+| PostgreSQL  | **✔️**  | **✔️**   | **✔️**‡  | **✔️**  | **❌**†    |
+| MySQL       | **✔️**  | **✔️**   | **✔️**   | **✔️**  | **❌**†    |
+| SQLite      | **✔️**  | **✔️**   | **✔️**‡  | **✔️**  | **❌**†    |
+| SQL Server  | **✔️**  | **✔️**   | **✔️**   | **✔️**  | **❌**†    |
+| CockroachDB | **✔️**  | **✔️**   | **✔️**   | **✔️**  | **❌**†    |
+| MongoDB     | **✔️**  | **✔️**   | **✔️**   | **✔️**  | **❌**†    |
+
+- † The `SetDefault` referential action is not supported.
+- ‡ The `NoAction` referential action is not supported for PostgreSQL and SQLite. Instead, use the `Restrict` action.
+
+Error messages returned by emulated referential actions in the `prisma` relation mode are generated by the Prisma Client and differ slightly from the error messages in the `foreignKeys` relation mode:
+
+```jsx
+Example:
+// foreignKeys:
+... Foreign key constraint failed on the field: `ProfileOneToOne_userId_fkey (index)`
+// prisma:
+... The change you are trying to make would violate the required relation 'ProfileOneToOneToUserOneToOne' between the `ProfileOneToOne` and `UserOneToOne` models.
+```
+
+### Introspection
+
+When you introspect a database with the `db pull` command with the `prisma` relation mode enabled, relations will not be automatically added to your schema. You will instead need to add any relations manually using the `@relation` attribute. This only needs to be done once – next time you introspect your database, Prisma will keep your added `@relation` attributes.
+
+### Prisma Migrate and <inlineCode>db push</inlineCode>
+
+When you apply changes to your Prisma schema with Prisma Migrate or `db push` with the `prisma` relation mode enabled, Prisma will not use foreign keys in your database.
+
+## Switch between relation modes
+
+It is only possible to switch between relation modes when you use a relational database connector (PostgreSQL, MySQL, SQLite, SQL Server, CockroachDB).
+
+### Switch from <inlineCode>foreignKeys</inlineCode> to <inlineCode>prisma</inlineCode>
```

Comment:
```suggestion
### Switch from <inlineCode>foreignKeys</inlineCode> (or default, no `relationMode` defined) to <inlineCode>prisma</inlineCode>
```

Reply:
It looks too clunky to me in the title, so I added it to the section itself (where it still looks pretty clunky, but maybe I can reword later)